packages/react/src/components/Stack/Stack.styles.ts

```diff
@@ -44,10 +47,9 @@ export const styles: IStackComponent['styles'] = (props, theme, tokens): IStackS
     textOverflow: 'ellipsis',
   };
 
-  // selectors to be applied regardless of wrap or direction
   const disableShrinkStyles = {
     // flexShrink styles are applied by the StackItem
-    '> *:not(.ms-StackItem)': {
+    [`> ${enableScopedSelectors ? '.' + GlobalClassNames.child : '*'}:not(.${StackItemGlobalClassNames.root})`]: {
```

Comment:
Maybe save the scoped selector to a local const, since you use it in multiple places. It would also make this easier to read.
```ts
const childSelector = enableScopedSelectors ? `> .${GlobalClassNames.child}` : '> *';
```
```suggestion
    [`${childSelector}:not(.${StackItemGlobalClassNames.root})`]: {
```